integration-app/src/main/java/uk/gov/crowncommercial/esourcing/app/ErrorResponse.java

```diff
@@ -0,0 +1,96 @@
+package uk.gov.crowncommercial.esourcing.app;
+
+import java.time.Instant;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+
+/**
+ * For returning customised JSON error responses. This mirrors the object returned by standard
+ * Spring Boot error handling but is a class and not simply a map of key value pairs.
+ */
+@JsonDeserialize(builder = ErrorResponse.Builder.class)
+public final class ErrorResponse {
+
+  private final Instant timestamp;
+  private final int status;
+  private final String error;
+  private final String message;
+  private final String path;
+
+  private ErrorResponse(Builder b) {
+    this.timestamp = b.timestamp;
+    this.status = b.status;
+    this.error = b.error;
+    this.message = b.message;
+    this.path = b.path;
+  }
+
+  public Instant getTimestamp() {
+    return timestamp;
+  }
+
+  public int getStatus() {
+    return status;
+  }
+
+  public String getError() {
+    return error;
+  }
+
+  public String getMessage() {
+    return message;
+  }
+
+  public String getPath() {
+    return path;
+  }
+
```

Comment:
Are these methods ever used?

Reply:
Only as get methods when serialising to JSON as part of the response.